diff --git a/view/frontend/web/js/overview.js b/view/frontend/web/js/overview.js
index 5d24370a29f62..540e09881881b 100644
--- a/view/frontend/web/js/overview.js
+++ b/view/frontend/web/js/overview.js
@@ -24,6 +24,14 @@ define([
          */
         _create: function () {
             this.element.on('submit', $.proxy(this._showLoader, this));
+            this._on(window, {
+                'pageshow': function (event) {
+                    if (event.originalEvent.persisted) {
+                        this.element.find(this.options.pleaseWaitLoader).hide().end()
+                            .find(this.options.placeOrderSubmit).prop('disabled', false).css('opacity', 1);
+                    }
+                }
+            });
         },
 
         /**
