diff --git a/view/frontend/web/js/newsletter-sign-up.js b/view/frontend/web/js/newsletter-sign-up.js
index 6c82ea435f5e4..ac859959fac4d 100644
--- a/view/frontend/web/js/newsletter-sign-up.js
+++ b/view/frontend/web/js/newsletter-sign-up.js
@@ -22,10 +22,18 @@ define([
 
         /** @inheritdoc */
         initialize: function (config, element) {
+            var self = this;
+
             this._super();
             this.element = element;
             $(element).on('change', $.proxy(this.updateSignUpStatus, this));
             this.updateSignUpStatus();
+
+            $(window).on('pageshow', function (event) {
+                if (event.originalEvent.persisted) {
+                    $(self.submitButton).prop('disabled', false);
+                }
+            });
         },
 
         /**
